addons/portal/views/portal_templates.xml

```diff
@@ -253,11 +253,15 @@
                             <div t-attf-class="form-group #{error.get('company_name') and 'has-error' or ''} col-lg-6">
                                 <label class="control-label label-optional" for="company_name">Company Name</label>
                                 <p t-if="partner.commercial_partner_id.is_company" class="form-control" t-esc="partner.commercial_company_name"/>
-                                <input t-else="" type="text" name="company_name" class="form-control" t-att-value="company_name or partner.company_name"/>
+                                <t t-else="1">
+                                    <input type="text" t-if="partner.can_edit_vat()" name="company_name" class="form-control" t-att-value="company_name or partner.company_name" />
+                                    <input type="text" t-else="1" readonly="1" name="company_name" class="form-control" t-att-value="company_name or partner.company_name" title="Changing company name is not allowed once document(s) have been issued for your account. Please contact us directly for this operation."/>
+                                </t>
                             </div>
-                            <div t-if="has_check_vat" t-attf-class="form-group #{error.get('vat') and 'has-error' or ''} col-lg-6">
```

Comment:
`has_check_vat` imho can be removed since it is not in checkout address step anyway
It will hide the vat field forever unless you install one of a few accounting localization that depends on `base_vat`